lib/spack/spack/cmd/create.py

```diff
@@ -681,8 +712,11 @@ def create(parser, args):
     build_system = get_build_system(args, guesser)
 
     # Create the package template object
+    constr_args = {'name': name, 'versions': versions}
     package_class = templates[build_system]
-    package = package_class(name, url, versions)
+    if package_class != BundlePackageTemplate:
+        constr_args['url'] = url
+    package = package_class(**constr_args)
```

Comment:
@tldahlgren I think this is causing https://github.com/spack/spack/issues/12529: none of the package classes take `**kwargs`, I think if you update them to do that, it will fix the issue.

Reply:
> @tldahlgren I think this is causing #12529: none of the package classes take `**kwargs`, I think if you update them to do that, it will fix the issue.

Ah.  Thanks Peter.